pkg/apis/minio.min.io/v2/helper_test.go

```diff
@@ -338,8 +338,8 @@ func TestTenant_GetDomainHosts(t1 *testing.T) {
 					Features: &Features{
 						Domains: &TenantDomains{
 							Minio: []string{
-								"https://domain1.com:8080",
```

Comment:
we need to support scheme in the domain list, it's how it's currently supported and with this change, we will break any user already using the domains feature

Reply:
In the latest commit, the code:
uses lcp to determine overlapping suffixes, as requested by @harshavardhana 
preserves url.parse, to avoid regressions as discussed with @dvaldivia 
keep original domain tests, to avoid regressions as discussed with @dvaldivia 
consider domain string my be prefixed by a schema in current setups, to avoid regressions as discussed with @dvaldivia